docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
+
+    let runAsync waitInSeconds =
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
+
+    // Output:
+    // Started Task
+    // Created Async
+    // Completed Task
+    // Completed Async
+
+### Async and Task Interop
+
+As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
+
+#### Async.AwaitTask
+
+This converts a Task into an async computation. It has the [signature](#functions-signatures): `Task<'T>` -> `Async<'T>`
+
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn $"File {fileName} has %d{bytes.Length} bytes"
+    }
+
+#### Async.StartAsTask
+
+This converts an async computation into a Task. It has the [signature](#functions-signatures): `Async<'T>` -> `Task<'T>`
+
+    async {
+        do! Async.Sleep 5000
+    } |> Async.StartAsTask
+
+### Async and the ThreadPool
+
+Below is a demonstration of the different ways to start an async computation and where it ends up in the dotnet runtime.
+In the comments; `M`, `X`, `Y`, and `Z` are used to represent differences in values.
+
+    #r "nuget:Nito.AsyncEx, 5.1.2"
+    
+    let asyncTask from =
+        async {
+            printfn $"{from} Thread Id = {System.Threading.Thread.CurrentThread.ManagedThreadId}"
+        }
+    
+    let run () =
+        printfn $"run() Thread Id = {System.Threading.Thread.CurrentThread.ManagedThreadId}"
+        asyncTask "StartImmediate" |> Async.StartImmediate      // run and wait on same thread
+        asyncTask "Start" |> Async.Start                        // queue in ThreadPool and do not wait
```

Comment:
I dont think I'd show this without explaining that you rarely want to randomly inject unawaited stuff into the threadpool in this way. It may well be an important part of explaining how everything works, but I would not have it in here as the third thing the example does